src/uct/ib/base/ib_md.c

```diff
@@ -74,6 +73,274 @@ static ucs_status_t uct_ib_md_query(uct_md_h uct_md, uct_md_attr_t *md_attr)
     return UCS_OK;
 }
 
+static ucs_status_t uct_ib_md_umr_qp_create(uct_ib_md_t *md) 
+{
+#if HAVE_EXP_UMR
+    struct ibv_exp_qp_init_attr qp_init_attr;
+    struct ibv_qp_attr qp_attr;
+    uint8_t port_num;
+    int ret;
+    uct_ib_device_t *ibdev;
+    struct ibv_exp_port_attr *port_attr;
+
+    md->umr_qp = NULL;
+    md->umr_cq = NULL;
+    ibdev = &md->dev;
+
+    if (!(ibdev->dev_attr.exp_device_cap_flags & IBV_EXP_DEVICE_UMR)) {
+        return UCS_ERR_UNSUPPORTED;
+    }
+
+    /* TODO: fix port selection. It looks like active port should be used */
+    port_num = 1;
+    port_attr = uct_ib_device_port_attr(ibdev, port_num);
+
+    memset(&qp_init_attr, 0, sizeof(qp_init_attr));
+
+    md->umr_cq = ibv_create_cq(ibdev->ibv_context, 1, NULL, NULL, 0);
+    if (md->umr_cq == NULL) {
+        ucs_error("failed to create UMR CQ: %m");
+        goto err;
+    }
+
+    qp_init_attr.qp_type             = IBV_QPT_RC;
+    qp_init_attr.send_cq             = md->umr_cq;
+    qp_init_attr.recv_cq             = md->umr_cq;
+    qp_init_attr.cap.max_inline_data = 0;
+    qp_init_attr.cap.max_recv_sge    = 1;
+    qp_init_attr.cap.max_send_sge    = 1;
+    qp_init_attr.srq                 = NULL;
+    qp_init_attr.cap.max_recv_wr     = 100;
+    qp_init_attr.cap.max_send_wr     = 100;
```

Comment:
Is there any particular reason why you use hardcoded values ? I would suggest to make it configurable